src/connection.rs

```diff
@@ -225,9 +216,12 @@ impl Connection {
         self.write_continuation = None;
         self.write_queue.clear();
         self.is_connected = false;
-        let timeout = ServerTimeout::Reconnect(self.token);
+        let timeout = ServerTimeout::Reconnect(token);
         let handle = event_loop.timeout_ms(timeout, duration).unwrap();
-        Ok((duration, timeout, handle))
+
+        info!("{:?}: {:?} reset, will attempt to reconnect in {}ms", self,
```

Comment:
Thinking about this more: I think it would be better to have a more structured way to log inside of connection.rs, while still retaining the server ID, but I don't want to block this patch from coming in because of this.  I think the thing to do for now is back out these specific changes from this patch, and I'll file a ticket to clean this up in the future.